packages/prosemirror-lwdita/src/commands.ts

```diff
@@ -177,6 +178,223 @@ export class InputContainer {
+ * @param node - Node selected node to edit
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function imageInputOverlay(callback: (args: any) => void, node?: Node): void {
+  // show the image upload dialog
+  // Create overlay
+  const overlay = document.createElement('div');
+  overlay.id = 'overlay';
+
+  // Create dialog
+  const dialog = document.createElement('div');
+  dialog.id = 'dialog';
+
+  // Create dialog content
+  const title = document.createElement('h1');
+  title.textContent = 'Upload Image';
+
+  const fileLabel = document.createElement('label');
+  fileLabel.textContent = 'File:';
+  fileLabel.htmlFor = 'fileInput';
+  const fileInput = document.createElement('input');
+  fileInput.type = 'file';
+  fileInput.id = 'fileInput';
+  fileInput.accept = 'image/*';
+
+  const urlLabel = document.createElement('label');
+  urlLabel.textContent = 'URL:';
+  urlLabel.htmlFor = 'urlInput';
+  const urlInput = document.createElement('input');
+  urlInput.type = 'text';
+  urlInput.id = 'urlInput';
+  urlInput.placeholder = 'url';
+
+  const embeddedLabel = document.createElement('label');
+  embeddedLabel.textContent = 'embed a copy:';
+  embeddedLabel.htmlFor = 'embeddedInput';
+  const embeddedInput = document.createElement('input');
+  embeddedInput.type = 'checkbox';
+  embeddedInput.id = 'embeddedInput';
+  embeddedInput.title = "this option will embed the image from an external source and store a copy in the document as base64 string"
+
+  const heightLabel = document.createElement('label');
+  heightLabel.textContent = 'Height:';
+  heightLabel.htmlFor = 'heightInput';
+  const heightInput = document.createElement('input');
+  heightInput.type = 'text';
+  heightInput.id = 'heightInput';
+  heightInput.placeholder = 'Height';
+
+  const widthLabel = document.createElement('label');
+  widthLabel.textContent = 'Width:';
+  widthLabel.htmlFor = 'widthInput';
+  const widthInput = document.createElement('input');
+  widthInput.type = 'text';
+  widthInput.id = 'widthInput';
+  widthInput.placeholder = 'Width';
+
+  const altTextLabel = document.createElement('label');
+  altTextLabel.textContent = 'Alt Text:';
+  altTextLabel.htmlFor = 'altTextInput';
+  const altTextInput = document.createElement('input');
+  altTextInput.type = 'text';
+  altTextInput.id = 'altTextInput';
+  altTextInput.placeholder = 'Alt Text';
+
+  const btnConatiner = document.createElement('div');
+  btnConatiner.id = 'btnConatiner';
+
+  const closeButton = document.createElement('button');
+  closeButton.id = 'closeButton';
+  closeButton.textContent = 'Dismiss';
+
+  const okButton = document.createElement('button');
+  okButton.id = 'okButton';
+  okButton.textContent = 'Insert';
+
+  btnConatiner.appendChild(closeButton);
+  btnConatiner.appendChild(okButton);
+
+  // Append dialog content to dialog
+  dialog.appendChild(title);
+
+  const fields = [
+    { label: fileLabel, input: fileInput },
```

Comment:
@marmoure We are currently not supporting all available LwDITA attributes for the image: 
Apart from `href`, `scope`, `height`, and `width`, the schema also allows to set `outputclass`, `format`, `keyref`, `dir`, and `xml:lang`.  For ensuring that this data from the source will be kept during the roundtrip, we could keep these values in `data-*` attributes.

Reply:
>  we could keep these values in data-* attributes.
There's no way of setting these values currently and I consider this out of scope.

Reply:
Ok.